### DISCUSSION TITLE
How to implement base classes in pyright-internal using a separate project?

### DISCUSSION THREAD

**Discussion by nchen63** [microsoft/pyright#3257] [Q&A] [answered]
Is there a recommended way to consume base classes in pyright-internal without forking? The published npm package is webpacked and doesn't have types.

**Comment by erictraut** [comment, marked as answer]
I think forking is the best approach. You will need to sync periodically. This is what we do with pylance, which builds upon pyright.

**Reply by nchen63** [reply]
Would it be possible to add a version number to pyright-root? I was able to install the tar.gz file as an npm dependency, but the install failed because the root package is missing a version number. I get: `error Can't add "pyright-root": invalid package version "".`

**Reply by erictraut** [reply]
I don't want to maintain a separate version number. You can use pyright version numbers or commit tags though.

**Reply by nchen63** [reply]
I'm not sure what you mean by "separate version number". Can't pyright-root use the same version number as pyright and pyright-internal?

**Reply by erictraut** [reply]
There are four package.json files I update each time I publish. I'm not interested in increasing the work involved in publishing for a niche use case. You can use the version from the pyright-internal package.json or use a fork.

**Reply by nchen63** [reply]
Ah, ok. I assumed there was a script that did the updates for the publish

**Reply by nchen63** [reply]
Just to make sure I'm not missing something, how would I use the pyright-internal package.json version number? When I `yarn add` the tar.gz file, it automatically tries to read the version number in the root directory and fails.
